### PR TITLE
Add new user option to CA1008

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Relates to [#5877](https://github.com/dotnet/roslyn-analyzers/pull/5877)

cc @mavasani

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1008.md [hunk 1]
```diff
@@ -50,11 +50,21 @@ Do not suppress a warning from this rule except for flags-attributed enumeration
 Use the following option to configure which parts of your codebase to run this rule on.
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Additional enum None names](#additional-enum-none-names)
 
 You can configure this option for just this rule, for all rules, or for all rules in this category (Design). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
 
 [!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]
 
+### Additional enum None names
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1008.md:59]: I'm afraid `None` may get localized while it shouldn't (applies to all occurrences of it)

Not sure if this should have backticks around or a no-loc.

Comment (Evangelink) [reply docs/fundamentals/code-analysis/quality-rules/ca1008.md:59]: Good remark, I don't know how this work for docs. @gewarren what's the guidance here please?

File: docs/fundamentals/code-analysis/quality-rules/ca1008.md [hunk 2]
```diff
@@ -50,11 +50,21 @@ Do not suppress a warning from this rule except for flags-attributed enumeration
...
+
+You can configure the list of additional allowed names for the enum None case (separated by `|`).
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1008.md:61]:
```suggestion
In .NET 7 and later versions, you can configure other allowable names for a zero-value enumeration field besides `None`. Separate multiple names by the `|` character. The following table shows some examples.
```

**Review by gewarren — COMMENTED** [review]
I think we should also modify the "Cause", "Description", and "How to fix violations" sections to include mention of the additional allowable names. E.g. under "Cause":

...defines a member that has a value of zero but its name is not 'None' or one of the [other allowable names](#additional-zero-value-field-names).

File: docs/fundamentals/code-analysis/quality-rules/ca1008.md [hunk 3]
```diff
@@ -50,11 +50,21 @@ Do not suppress a warning from this rule except for flags-attributed enumeration
...
+
+| Option value | Summary |
+| --- | --- |
+| `dotnet_code_quality.CA1008.additional_enum_none_names = Never` | Allows both `None` and `Never` |
+| `dotnet_code_quality.CA1008.additional_enum_none_names = Never|Nothing` | Allows `None`, `Never` and `Nothing` |
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply docs/fundamentals/code-analysis/quality-rules/ca1008.md:66]: I always forget that in English we should put comma before `and`, thanks for the reminder :)

[the remark this answers is not in the document]
Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca1008.md]: Well that is just the style of choice for Microsoft. This style (called Oxford comma) is actually quite contentious even among English speakers. For example, New York Times does not used the Oxford comma.

File: docs/fundamentals/code-analysis/quality-rules/ca1008.md [hunk 4]
```diff
@@ -50,11 +50,21 @@ Do not suppress a warning from this rule except for flags-attributed enumeration
 Use the following option to configure which parts of your codebase to run this rule on.
 
 - [Include specific API surfaces](#include-specific-api-surfaces)
+- [Additional enum None names](#additional-enum-none-names)
```
Comment (Evangelink) [inline docs/fundamentals/code-analysis/quality-rules/ca1008.md:53]:
```suggestion
- [Additional enum None names](#additional-zero-value-field-names)
```

**Review by gewarren — APPROVED** [review]
Thank you @Evangelink!

**Pull request merged**